api/src/main/java/org/pmiops/workbench/api/BugReportController.java

```diff
@@ -1,119 +0,0 @@
-package org.pmiops.workbench.api;
-
-import com.google.common.annotations.VisibleForTesting;
-import com.google.common.collect.ImmutableList;
-import java.io.File;
-import java.io.FileOutputStream;
-import java.io.IOException;
-import java.util.List;
-import java.util.Optional;
-import java.util.logging.Logger;
-import javax.inject.Provider;
-import org.pmiops.workbench.db.model.User;
-import org.pmiops.workbench.exceptions.BadRequestException;
-import org.pmiops.workbench.exceptions.ServerErrorException;
-import org.pmiops.workbench.jira.JiraService;
-import org.pmiops.workbench.model.BillingProjectStatus;
-import org.pmiops.workbench.model.BugReport;
-import org.pmiops.workbench.notebooks.ApiException;
-import org.pmiops.workbench.notebooks.NotebooksService;
-import org.pmiops.workbench.notebooks.api.JupyterApi;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.RestController;
-
-@RestController
-public class BugReportController implements BugReportApiDelegate {
-  private static final Logger log = Logger.getLogger(BugReportController.class.getName());
-  private static final List<String> notebookLogFiles =
-      ImmutableList.of("delocalization.log", "jupyter.log", "localization.log");
-
-  private final Provider<JiraService> jiraServiceProvider;
-  private final Provider<JupyterApi> jupyterApiProvider;
-  private Provider<User> userProvider;
-
-  @Autowired
-  BugReportController(
-      Provider<User> userProvider,
-      Provider<JupyterApi> jupyterApiProvider,
-      Provider<JiraService> jiraService) {
-    this.userProvider = userProvider;
-    this.jupyterApiProvider = jupyterApiProvider;
-    this.jiraServiceProvider = jiraService;
-  }
-
-  @VisibleForTesting
-  void setUserProvider(Provider<User> userProvider) {
-    this.userProvider = userProvider;
-  }
-
-
-  @Override
-  public ResponseEntity<BugReport> sendBugReport(BugReport bugReport) {
-    User user = userProvider.get();
-    JupyterApi jupyterApi = jupyterApiProvider.get();
-    JiraService jiraService = jiraServiceProvider.get();
-    try {
-      String issueKey = jiraService.createIssue(bugReport).getKey();
-      // If requested, try to pull logs from the notebook cluster using the researcher's creds. Some
-      // or all of these log files might be missing, or the cluster may not even exist, so ignore
-      // failures here.
-      if (Optional.ofNullable(bugReport.getIncludeNotebookLogs()).orElse(false) &&
-          BillingProjectStatus.READY.equals(user.getFreeTierBillingProjectStatusEnum())) {
-        for (String fileName : BugReportController.notebookLogFiles) {
-          try {
-            String logContent = jupyterApi.getRootContents(
```

Comment:
i'll file a ticket

Reply:
+1, I wonder if it will be simple enough to give users a button to download the current cluster's logs (and upload them via Zendesk if asked to share).

Reply:
that's exactly what i was thinking - https://precisionmedicineinitiative.atlassian.net/browse/RW-2446

might try and do that now as a quick follow up while some of the old code to do so is still fresh